scripts/install.sh

```diff
@@ -1,13 +1,22 @@
 #!/bin/bash
 
+REPO="https://github.com/software-mansion/universal-sierra-compiler"
 BINARY_NAME="universal-sierra-compiler"
 LOCAL_BIN="${HOME}/.local/bin"
 
 main () {
   check_cmd curl
   check_cmd tar
 
-  download_and_extract_binary
+  version=${1:-latest}
+  release_tag=$(curl -# -Ls -H 'Accept: application/json' "${REPO}/releases/{$version}" | sed -e 's/.*"tag_name":"\([^"]*\)".*/\1/')
+
+  if [[ $release_tag == *"error"* ]]; then
+    echo "Please pass correct version (e.g. \"v1.0.0\")"
+    exit 1
+  fi
```

Comment:
nit: you can proabbly add `--fail` flag to curl, and then this `if` can just become `[ -z "$release_tag" ]`
the error message could also be improved, to something like "No such version {ver}, please pass correct version ..."